### PR TITLE
Teams

### PR DESCRIPTION
**Pull request by sarimrmalik**
- [x] Create [data model](https://whimsical.com/data-model-SN6cLKLRfRDPYhnJoJStkb)
- [x] Add prisma adapter to handle authentication
- [x] Add support for teams

### CODE REVIEW

File: lib/components/dashboard/Buttons/Sun.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {cn} from '~/utils'
 export const Sun = ({className}: {className?: string}) => {
 	return (
 		<Button
-			onClick={() => signIn('github', {callbackUrl: '/dashboard'})}
```
Comment (tedspare) [inline lib/components/dashboard/Buttons/Sun.tsx:9]: Why does this redirect to `/` after login now? Does the `/` path have its own redirect logic?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -72,11 +82,102 @@ model Usage {
...
+
+model Team {
+  id        String   @id @default(cuid())
+  slug      String   @unique
+  name      String?
+  createdBy String
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  user       User         @relation(fields: [createdBy], references: [id])
```
Comment (tedspare) [inline prisma/schema.prisma:141]: Should this be `admin` for clarity?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -72,11 +82,102 @@ model Usage {
...
+  expires    DateTime
+
+  @@unique([identifier, token])
+}
+
+model Team {
+  id        String   @id @default(cuid())
+  slug      String   @unique
+  name      String?
+  createdBy String
```
Comment (tedspare) [inline prisma/schema.prisma:137]: Should this be `createdBy String **@id**`?

File: lib/components/dashboard/Buttons/Sun.tsx (hunk 1 above)
Comment (sarimrmalik) [reply lib/components/dashboard/Buttons/Sun.tsx:9]: We don't have a `/dashboard` route anymore, it's a dynamic route which means we check for session on the home page `/` and redirect to the relevant `/[team]`,

File: prisma/schema.prisma (hunk 3 above)
Comment (sarimrmalik) [reply prisma/schema.prisma:166]: The admin can change over time but the creator does not. The Admin relationship is captured in the `Membership` table.

File: prisma/schema.prisma (hunk 2 above)
Comment (sarimrmalik) [reply prisma/schema.prisma]: This isn't always the admin, renamed this to `creator` for clarity.

**Comment by DexterStorey** [comment]
LGTM 🚀

**Review by tedspare — APPROVED** [review]
LGTM

**Pull request merged**